fnet/network_test.go

```diff
@@ -53,3 +53,42 @@ func TestNormalizePort(t *testing.T) {
 		}
 	}
 }
+
+func TestGRPCDestination(t *testing.T) {
+	tests := []struct {
+		name   string
+		dest   string
+		port   string
+		output string
+	}{
+		{
+			"port number only",
+			"localhost",
+			"8079",
+			"localhost:8079",
+		},
+		{
+			"IPv4 address",
+			"1.2.3.4",
+			"8078",
+			"1.2.3.4:8078",
+		},
+		{
+			"IPv6 address",
+			"2001:dba::1",
+			"8079",
+			"[2001:dba::1]:8079",
+		},
```

Comment:
I would add a test
NormalizeHostPort([2001:dba::1]:1234,4567) -> [2001:dba::1]:1234 because there is already a port in 'host'
and
NormalizeHostPort(2001:dba::1234,4567) -> [2001:dba::1234]:4567 because the : is for ipv6 and not a port
and
NormalizeHostPort(foo.bar.com:777,888) -> foo.bar.com:777
NormalizeHostPort(foo.bar.com,888) -> foo.bar.com:888

Reply:
`NormalizeHostPort([2001:dba::1]:1234,4567) -> [2001:dba::1]:1234` will fail:
```
--- FAIL: TestNormalizeHostPort (0.00s)
	network_test.go:93: Test case IPv6 address and port failed to set gRPC destination
			expected: [2001:dba::1]:1234
			  actual: [2001:dba::1]:1234:5678
```

The func expects host to be an IP address and not an ip:port pair. Do you want the func to support this use case or should I update the test to support expected failures?

Reply:
Those tests should pass (which is why I mentioned to add the test and then fix the code)

I think again you can do a string based

Reply:
can you add those 2:

NormalizeHostPort(foo.bar.com:777,888) -> foo.bar.com:777
NormalizeHostPort(foo.bar.com,888) -> foo.bar.com:888